Source/WebKit/UIProcess/Extensions/WebExtension.h

```diff
@@ -192,79 +202,78 @@ class WebExtension : public API::ObjectImpl<API::Object::Type::WebExtension>, pu
     bool supportsManifestVersion(double version) { ASSERT(version > 2); return manifestVersion() >= version; }
 
     Ref<API::Data> serializeLocalization();
 
+#if PLATFORM(COCOA)
     SecStaticCodeRef bundleStaticCode() const;
     NSData *bundleHash() const;
 
-#if PLATFORM(MAC)
     bool validateResourceData(NSURL *, NSData *, NSError **);
+
+    UTType *resourceTypeForPath(StringView);
+    NSLocale *defaultLocale();
 #endif
 
     bool isWebAccessibleResource(const URL& resourceURL, const URL& pageURL);
 
-    NSURL *resourceFileURLForPath(NSString *);
-
-    UTType *resourceTypeForPath(NSString *);
+    URL *resourceFileURLForPath(StringView);
 
-    NSString *resourceStringForPath(NSString *, CacheResult = CacheResult::No, SuppressNotFoundErrors = SuppressNotFoundErrors::No);
-    NSData *resourceDataForPath(NSString *, CacheResult = CacheResult::No, SuppressNotFoundErrors = SuppressNotFoundErrors::No);
+    String resourceStringForPath(String, CacheResult = CacheResult::No, SuppressNotFoundErrors = SuppressNotFoundErrors::No);
+    std::span<const uint8_t> resourceDataForPath(String, CacheResult = CacheResult::No, SuppressNotFoundErrors = SuppressNotFoundErrors::No);
 
     _WKWebExtensionLocalization *localization();
-    NSLocale *defaultLocale();
-
-    NSString *displayName();
-    NSString *displayShortName();
-    NSString *displayVersion();
-    NSString *displayDescription();
-    NSString *version();
+    String displayName();
+    String displayShortName();
+    String displayVersion();
+    String displayDescription();
+    String version();
 
-    NSString *contentSecurityPolicy();
+    String contentSecurityPolicy();
 
-    CocoaImage *icon(CGSize idealSize);
+    RefPtr<WebCore::Icon> icon(WebCore::FloatSize idealSize);
 
-    CocoaImage *actionIcon(CGSize idealSize);
-    NSString *displayActionLabel();
-    NSString *actionPopupPath();
+    RefPtr<WebCore::Icon> actionIcon(WebCore::FloatSize idealSize);
+    String displayActionLabel();
+    String actionPopupPath();
 
     bool hasAction();
     bool hasBrowserAction();
     bool hasPageAction();
 
 #if ENABLE(WK_WEB_EXTENSIONS_SIDEBAR)
-    CocoaImage *sidebarIcon(CGSize idealSize);
-    NSString *sidebarDocumentPath();
-    NSString *sidebarTitle();
+    RefPtr<WebCore::Icon> sidebarIcon(WebCore::FloatSize idealSize);
+    String sidebarDocumentPath();
+    String sidebarTitle();
 #endif
 
-    CocoaImage *imageForPath(NSString *);
+    RefPtr<WebCore::Icon> imageForPath(String);
 
-    NSString *pathForBestImageInIconsDictionary(NSDictionary *, size_t idealPixelSize);
-    CocoaImage *bestImageInIconsDictionary(NSDictionary *, size_t idealPointSize);
-    CocoaImage *bestImageForIconsDictionaryManifestKey(NSDictionary *, NSString *manifestKey, CGSize idealSize, RetainPtr<CocoaImage>& cacheLocation, Error, NSString *customLocalizedDescription);
+    String pathForBestImageInIconsDictionary(RefPtr<JSON::Value> value, size_t idealPixelSize);
```

Comment:
For all of these `RefPtr<JSON::Value>`'s, if any of them should never be `nullptr`, then use `const JSON::Value&`